crates/evm/traces/src/lib.rs

```diff
@@ -59,193 +48,24 @@ pub enum DecodedCallLog<'a> {
-            &mut s,
-            "{}{} {}@{}",
-            CALL.yellow(),
-            "new".yellow(),
-            decoded.label.as_deref().unwrap_or("<unknown>"),
-            address
-        )?;
-    } else {
-        let (func_name, inputs) = match &decoded.func {
-            Some(DecodedCallData { signature, args }) => {
-                let name = signature.split('(').next().unwrap();
-                (name.to_string(), args.join(", "))
-            }
-            None => {
-                debug!(target: "evm::traces", trace=?trace, "unhandled raw calldata");
-                if trace.data.len() < 4 {
-                    ("fallback".to_string(), hex::encode(&trace.data))
-                } else {
-                    let (selector, data) = trace.data.split_at(4);
-                    (hex::encode(selector), hex::encode(data))
-                }
-            }
-        };
-
-        let action = match trace.kind {
-            CallKind::Call => "",
-            CallKind::StaticCall => " [staticcall]",
-            CallKind::CallCode => " [callcode]",
-            CallKind::DelegateCall => " [delegatecall]",
-            CallKind::Create | CallKind::Create2 => unreachable!(),
-            CallKind::AuthCall => " [authcall]",
-        };
-
-        let color = trace_color(trace);
-        write!(
-            &mut s,
-            "{addr}::{func_name}{opt_value}({inputs}){action}",
-            addr = decoded.label.as_deref().unwrap_or(&address).fg(color),
-            func_name = func_name.fg(color),
-            opt_value = if trace.value.is_zero() {
-                String::new()
-            } else {
-                format!("{{value: {}}}", trace.value)
-            },
-            action = action.yellow(),
-        )?;
-    }
-
-    Ok((s, decoded.return_data))
-}
-
-/// Render a trace log.
-async fn render_trace_log(
-    log: &LogData,
-    decoder: &CallTraceDecoder,
-) -> Result<String, std::fmt::Error> {
-    let mut s = String::new();
-    let decoded = decoder.decode_event(log).await;
-
-    match decoded {
-        DecodedCallLog::Raw(log) => {
-            for (i, topic) in log.topics().iter().enumerate() {
-                writeln!(
-                    s,
-                    "{:>13}: {}",
-                    if i == 0 { "emit topic 0".to_string() } else { format!("topic {i}") },
-                    format!("{topic:?}").cyan()
-                )?;
-            }
-
-            write!(s, "          data: {}", hex::encode_prefixed(&log.data).cyan())?;
-        }
-        DecodedCallLog::Decoded(name, params) => {
-            let params = params
-                .iter()
-                .map(|(name, value)| format!("{name}: {value}"))
-                .collect::<Vec<String>>()
-                .join(", ");
-
-            write!(s, "emit {}({params})", name.cyan())?;
-        }
-    }
+/// Render a collection of call traces.
+pub async fn render_trace_arena(arena: &mut CallTraceArena) -> Result<String, std::fmt::Error> {
```

Comment:
unnecessary async and result

Reply:
good call, updated to be not async

decoded_traces's `join_all` should still be concurrent with the changes